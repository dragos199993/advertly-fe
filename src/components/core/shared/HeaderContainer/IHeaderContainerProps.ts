import { ReactElement } from 'react';

export interface IHeaderContainerProps {
  children: (props: IHeaderProps) => ReactElement<IHeaderProps>;
}

export interface IHeaderContainerState {
  isNavbarOpen: boolean;
}

export interface IHeaderProps extends IHeaderContainerState {
  toggleNavbarMenu: () => void;
}
