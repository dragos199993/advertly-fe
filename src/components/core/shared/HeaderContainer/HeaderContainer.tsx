import { Component, ReactNode } from 'react';
import { IHeaderContainerProps, IHeaderContainerState } from './IHeaderContainerProps';

class HeaderContainer extends Component<IHeaderContainerProps, IHeaderContainerState> {
  state: IHeaderContainerState = {
    isNavbarOpen: false
  };

  toggleNavbarMenu = () => {
    this.setState({
      isNavbarOpen: !this.state.isNavbarOpen
    });
  };

  render(): ReactNode {
    const { children } = this.props;
    return children({
      toggleNavbarMenu: this.toggleNavbarMenu,
      isNavbarOpen: this.state.isNavbarOpen
    });
  }
}

export default HeaderContainer;
