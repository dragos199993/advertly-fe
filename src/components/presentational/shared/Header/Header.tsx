import React, { FunctionComponent } from 'react';
import {
  Collapse,
  Navbar,
  NavbarToggler,
  NavbarBrand,
  Nav,
  NavItem,
  NavLink,
  UncontrolledDropdown,
  DropdownToggle,
  DropdownMenu,
  DropdownItem
} from 'reactstrap';
import { IHeaderProps } from '../../../core/shared/HeaderContainer/IHeaderContainerProps';

const Header: FunctionComponent<IHeaderProps> = ({ isNavbarOpen, toggleNavbarMenu }) => {
  return (
    <div>
      <Navbar color="light" light={ true } expand="md">
        <NavbarBrand href="/">nedsilon blog</NavbarBrand>
        <NavbarToggler onClick={ toggleNavbarMenu } />
        <Collapse isOpen={ isNavbarOpen } navbar={ true }>
          <Nav className="ml-auto" navbar={ true }>
            <NavItem>
              <NavLink href="/components/">Components</NavLink>
            </NavItem>
            <NavItem>
              <NavLink href="https://github.com/reactstrap/reactstrap">GitHub</NavLink>
            </NavItem>
            <UncontrolledDropdown nav={ true } inNavbar={ true }>
              <DropdownToggle nav={ true } caret={ true }>
                Options
              </DropdownToggle>
              <DropdownMenu right={ true }>
                <DropdownItem>
                  Option 1
                </DropdownItem>
                <DropdownItem>
                  Option 2
                </DropdownItem>
                <DropdownItem divider={ true } />
                <DropdownItem>
                  Reset
                </DropdownItem>
              </DropdownMenu>
            </UncontrolledDropdown>
          </Nav>
        </Collapse>
      </Navbar>
    </div>
  );
};

export default Header;
