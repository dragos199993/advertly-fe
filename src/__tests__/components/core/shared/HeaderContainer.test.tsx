import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import HeaderContainer from '../../../../components/core/shared/HeaderContainer/HeaderContainer';
import Header from '../../../../components/presentational/shared/Header/Header';
import {
  IHeaderContainerProps,
  IHeaderContainerState
} from '../../../../components/core/shared/HeaderContainer/IHeaderContainerProps';

describe('HeaderContainer', () => {
  let wrapperContainer: ShallowWrapper<IHeaderContainerProps, IHeaderContainerState>;
  wrapperContainer = shallow<IHeaderContainerProps, IHeaderContainerState>(
    <HeaderContainer>
      { (props) => <Header { ...props } /> }
    </HeaderContainer>
  );

  test('renders without crashing', () => {
    expect(wrapperContainer).toMatchSnapshot();
  });
});
