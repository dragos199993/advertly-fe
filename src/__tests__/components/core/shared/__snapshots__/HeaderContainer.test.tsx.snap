// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeaderContainer>
    [Function]
  </HeaderContainer>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "isNavbarOpen": false,
      "toggleNavbarMenu": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "isNavbarOpen": false,
        "toggleNavbarMenu": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
